src/main/java/seedu/address/model/Model.java

```diff
@@ -19,14 +20,28 @@
     void resetData(ReadOnlyTaskBook newData);
 
     /** Returns the TaskBook */
-    ReadOnlyTaskBook getAddressBook();
+    ReadOnlyTaskBook getTaskBook();
 
     /** Deletes the given task. */
     void deleteTask(Task target) throws TaskNotFoundException;
 
     /** Adds the given task */
     void addTask(Task task);
 
+    //undo and redo
+    /** Undo prev action that modifies data
+     * @throws IllegalValueException **/
```

Comment:
We can easily see that this throws an `IllegalValueException`, so this comment on its own adds no value.

You need to explain *when* this exception is thrown.